### PR TITLE
chore: adopt @nodevu/core and remove node-version-data

### PR DESCRIPTION
**Pull request by ovflowd**
This PR removes the outdated `node-version-data` package in favour of using modern Node APIS (`@nodevu`).

The package is maintained by @bnb and is getting into `@pkgjs` (if I recall correctly).

This also fixes some of the issues we have with the current library.

### CODE REVIEW

**Comment by tniessen** [comment]
> This also fixes some of the issues we have with the current library.

Could you elaborate on said issues?

**Comment by ovflowd** [comment]
> Could you elaborate on said issues?

- Being CommonJS whilst the remaining codebase of Node.js org ESM
- Being slow (relatively compared to `@nodevu`
- Unmaintained
- When API requests fails there's a break of API (without throwing errors) causing unrelated issues within the build of the Website

**Comment by richardlau** [comment]
> This PR removes the outdated `node-version-data` package in favour of using modern Node APIS (`@nodevu`).
> 

Are you using anything from that that isn't covered by https://github.com/pkgjs/nv?

**Comment by ovflowd** [comment]
Ye, such as dependencies (npm) and their versions.

The proposed package also has an elegant way of showing some meta information like support information (is security release, lts release) what are the latest security releases for a given major and many other metadata that can definitely be helpful in the future.

File: next.config.mjs [hunk 1]
```diff
@@ -23,4 +23,5 @@ export default withNextra({
   images: { unoptimized: true },
   outputFileTracing: false,
   basePath: process.env.NEXT_BASE_PATH || '',
+  experimental: { mdxRs: true },
```
Comment (mikeesto) [inline next.config.mjs]: Is this related to swapping to nodevu?

Comment (ovflowd) [reply next.config.mjs]: Not at all! This should be removed lol. It also has 0 effect as nextra uses its own mdx stuff

**Comment by ovflowd** [comment]
@richardlau @tniessen are you folks OK with the change?

cc @nodejs/website for approvals / review

**Comment by ovflowd** [comment]
A friendly bump @nodejs/website

**Review by benhalverson — APPROVED** [review]
Cookie Monster

**Comment by ovflowd** [comment]
Ty @benhalverson, glad to see you around 🙇 ❤️

**Review by aymen94 — APPROVED** [review]

**Review by mikeesto — APPROVED** [review]
